spec/abci++/abci++_basic_concepts_002_draft.md

```diff
@@ -1,51 +1,266 @@
+
+### Info methods
+
+For initialization and queries from the user.
+
+* [**Info:**](./abci++_methods_002_draft.md#info) Used to sync Tendermint with the Application during a handshake that happens on startup.
+
+* [**Query:**](./abci++_methods_002_draft.md#query) Clients can use this method to query the Application for information about the application state.
+
+### State-sync methods
+
+State sync allows new nodes to rapidly bootstrap by discovering, fetching, and applying
+state machine snapshots instead of replaying historical blocks. For more details, see the
+[state sync section](../p2p/messages/state-sync.md).
+
+New nodes will discover and request snapshots from other nodes in the P2P network.
+A Tendermint node that receives a request for snapshots from a peer will call
+`ListSnapshots` on its Application to retrieve any local state snapshots. After receiving
+ snapshots from peers, the new node will offer each snapshot received from a peer
+to its local Application via the `OfferSnapshot` method.
+
+Snapshots may be quite large and are thus broken into smaller "chunks" that can be
+assembled into the whole snapshot. Once the Application accepts a snapshot and
+begins restoring it, Tendermint will fetch snapshot "chunks" from existing nodes.
+The node providing "chunks" will fetch them from its local Application using
+the `LoadSnapshotChunk` method.
+
+As the new node receives "chunks" it will apply them sequentially to the local
+application with `ApplySnapshotChunk`. When all chunks have been applied, the
+Application's `AppHash` is retrieved via an `Info` query. The `AppHash` is then
+compared to the blockchain's `AppHash` which is verified via
+[light client verification](../light-client/verification/README.md).
+
+In summary:
+
+* [**ListSnapshots:**](./abci++_methods_002_draft.md#listsnapshots) Used by nodes to discover available snapshots on peers.
+
+* [**LoadSnapshotChunk:**](./abci++_methods_002_draft.md#loadsnapshotchunk) Used by nodes to retrieve snapshot chunks from peers.
+
+* [**OfferSnapshot:**](./abci++_methods_002_draft.md#offersnapshot) When a node receives a snapshot from a peer, Tendermint uses this method to offer the snapshot to the Application.
+
+* [**ApplySnapshotChunk:**](./abci++_methods_002_draft.md#applysnapshotchunk) Used by Tendermint to hand snapshot chunks to the Application.
+
+### Other methods
+
+Additionally, there is a [**Flush**](./abci++_methods_002_draft.md#flush) method that is called on every connection,
+and an [**Echo**](./abci++_methods_002_draft.md#echo) method that is just for debugging.
 
 >**TODO** Figure out what to do with this.
```

Comment:
Hmmm. I don't remember that this TODO refers to... do you? If not, please remove